### PR TITLE
Clarify `uses` node in typed fragments

### LINKED ISSUE #718: Clarify that libraries may be used in any type of fragment
The spec says:
> Any number of libraries can be applied by using the OPTIONAL uses node ONLY at the root of a master RAML or RAML fragment file.

(see ["Applying Librairies"](https://github.com/raml-org/raml-spec/blob/master/versions/raml-10/raml-10.md#applying-libraries))

However the `uses` property is only listed under the "Library" fragment list of properties.

### Proposed changes
- in the "[Typed Fragments](https://github.com/raml-org/raml-spec/blob/master/versions/raml-10/raml-10.md#typed-fragments)" section, underneath the fragment identifiers table, add a table of allowed nodes containing `uses?`, e.g. `In addition to any nodes allowed in any of the Relevant RAML Specification Section above, the following OPTIONAL nodes are allowed:`
- add `typed` before each occurrence of `fragment file` (as originally suggested in #357)
- remove the word `master` to avoid confusion, i.e. `root of a RAML or RAML fragment file` (the term "master" is only used in the context of Overlays & Extensions)
- remove the unnecessary word "ONLY", i.e. `Any number of libraries can be applied by using the OPTIONAL uses node at the root of a RAML or RAML typed fragment file`

### PR DESCRIPTION
**Pull request by jstoiko**
Closes #718

### CODE REVIEW

**Comment by BlockLucas** [comment]
Hi! 

Typed fragments didn't specify the usage of “uses” facet, hence the explanation that they MUST be valid according to the relevant RAML specification forbids that.
For us is not a PATCH change as a NamedExample with "uses" as defined in 1.0.1 would be invalid for 1.0

**Comment by jstoiko** [comment]
@BlockLucas: I think you're putting the finger on the issue that led us to making this clarification (and other clarifications we're including in 1.0.1). Libraries were always meant to be used within any kind of typed fragment. This is what the RAML 1.0 Spec currently says:

> Any number of libraries can be applied by using the OPTIONAL **uses** node ONLY at the root of a master RAML or **RAML fragment file**.
> (...)
> For example, **a RAML type fragment file that uses a library** of remote types can import one of those types by referring to it. The remote type is included as if it were defined locally within the RAML type fragment file.

What we did is we went back to the original discussions that took place prior to the release of the RAML 1.0 spec to understand what the original intent was. And in this case it was pretty obvious that the intent has always been to have libraries be used in any RAML file, whether a RAML API or RAML typed fragment definition. The fact that the "Typed Fragments" section does not have the `uses:` node facet defined currently is an obvious mistake. Fixing this mistake by adding this table doesn't change the fact that these two sentences above were there in the first place.

File: versions/raml-10/raml-10.md [hunk 1]
```diff
@@ -3239,11 +3245,11 @@ resourceTypes:
 
 #### Applying Libraries
 
-Any number of libraries can be applied by using the OPTIONAL **uses** node ONLY at the root of a master RAML or RAML fragment file. The value of the uses node is a map of key-value pairs. The keys are treated as library names, or namespaces, and the value MUST be the location of a RAML library file, usually an external RAML library fragment document.
+Any number of libraries can be applied by using the OPTIONAL **uses** node at the root of a RAML or RAML typed fragment file. The value of the uses node is a map of key-value pairs. The keys are treated as library names, or namespaces, and the value MUST be the location of a RAML library file, usually an external RAML library fragment document.
```
Comment (priyaayyar) [inline versions/raml-10/raml-10.md:3248]:
```suggestion
You can apply any number of libraries by using the OPTIONAL `uses` node at the root of a RAML or RAML typed fragment file. The value of the `uses` node is a map of the key-value pairs. The keys are treated as library names, or namespaces, and the value must be the location of a RAML library file, usually an external RAML library fragment document.
```

File: versions/raml-10/raml-10.md [hunk 2]
```diff
@@ -3239,11 +3245,11 @@ resourceTypes:
...
 
-Using **uses** does not automatically import the remote library assets into the local file, but the local file can import those assets by referring to the assets from its contents. For example, a RAML type fragment file that uses a library of remote types can import one of those types by referring to it. The remote type is included as if it were defined locally within the RAML type fragment file.
+Using **uses** does not automatically import the remote library assets into the local file, but the local file can import those assets by referring to the assets from its contents. For example, a RAML typed fragment file that uses a library of remote types can import one of those types by referring to it. The remote type is included as if it were defined locally within the RAML typed fragment file.
```
Comment (priyaayyar) [inline versions/raml-10/raml-10.md:3252]:
```suggestion
The `uses` node does not automatically import the remote library assets into the local file. However, the local file can import those assets by referring to the assets from its contents. For example, a RAML typed fragment file that uses a library of remote types can import one of those types by referring to it. The remote type is included as if it were defined locally within the RAML typed fragment file.
```

File: versions/raml-10/raml-10.md [hunk 3]
```diff
@@ -3074,9 +3074,15 @@ A file to be included MAY begin with a RAML fragment identifier line, which cons
 | Extension | An extension file | [Extensions](#extensions)
 | SecurityScheme | A definition of a security scheme | [Security Schemes](#security-schemes)
 
-If a file begins with a RAML fragment identifier line, and the fragment identifier is not Library, Overlay, or Extension, the contents of the file after removal of the RAML fragment identifier line MUST be valid structurally according to the relevant RAML specification. For example, a RAML file beginning with `#%RAML 1.0 Trait` must have the structure of a RAML trait declaration as defined in the [Resource Types and Traits](#resource-types-and-traits) section. Including the file in a correct location for a trait declaration results in a valid RAML file.
+In addition to any nodes allowed in any of the "Relevant RAML Specification Sections" above, the following OPTIONAL node is allowed:
```
Comment (priyaayyar) [inline versions/raml-10/raml-10.md:3077]:
```suggestion
In addition to the nodes allowed in any of the Relevant RAML Specification Sections above, the following optional node is allowed:
```

File: versions/raml-10/raml-10.md [hunk 4]
```diff
@@ -3074,9 +3074,15 @@ A file to be included MAY begin with a RAML fragment identifier line, which cons
...
 
-The following example shows a RAML fragment file that defines a resource type and a file that includes this fragment file.
+|Allowed Node | Description |
+|:------------|:------------|
+| uses? | Imported external [libraries](#libraries) for use within the RAML typed fragment. |
```
Comment (priyaayyar) [inline versions/raml-10/raml-10.md:3081]: Is the question mark uses? intentional?

File: versions/raml-10/raml-10.md [hunk 5]
```diff
@@ -3074,9 +3074,15 @@ A file to be included MAY begin with a RAML fragment identifier line, which cons
...
+
+If a file begins with a RAML fragment identifier line, and the fragment identifier is not Library, Overlay, or Extension, the contents of the file after removal of the RAML fragment identifier line and any `uses` node MUST be valid structurally according to the relevant RAML specification. For example, a RAML file beginning with `#%RAML 1.0 Trait` must have the structure of a RAML trait declaration as defined in the [Resource Types and Traits](#resource-types-and-traits) section. Including the file in a correct location for a trait declaration results in a valid RAML file.
```
Comment (priyaayyar) [inline versions/raml-10/raml-10.md:3083]:
```suggestion
If a file begins with a RAML fragment identifier line, and the fragment identifier is not of type Library, Overlay, or Extension, the contents of the file after removal of the RAML fragment identifier line and any `uses` node must be valid structurally according to the relevant RAML specification. For example, a RAML file beginning with `#%RAML 1.0 Trait` must have the structure of a RAML trait declaration as defined in the [Resource Types and Traits](#resource-types-and-traits) section. Including the file in a correct location for a trait declaration results in a valid RAML file.
```

File: versions/raml-10/raml-10.md [hunk 6]
```diff
@@ -3074,9 +3074,15 @@ A file to be included MAY begin with a RAML fragment identifier line, which cons
...
+
+The following example shows a RAML fragment file that defines a resource type and a file that includes this typed fragment file.
```
Comment (priyaayyar) [inline versions/raml-10/raml-10.md:3085]:
```suggestion
The following example shows a RAML fragment file that defines a resource type and a file that includes (this o the?) typed fragment file.
```

File: versions/raml-10/raml-10.md [hunk 7]
```diff
@@ -3206,11 +3212,11 @@ traits:
 
 RAML libraries are used to combine any collection of data type declarations, resource type declarations, trait declarations, and security scheme declarations into modular, externalized, reusable groups. While libraries are intended to define common declarations in external documents, which are then included where needed, libraries can also be defined inline.
 
-The following table describes the nodes, which are optional, of a library node.
+In addition to any nodes allowed at the root of [RAML typed fragments](#typed-fragments), RAML libraries allow the following optional nodes:
```
Comment (priyaayyar) [inline versions/raml-10/raml-10.md:3215]:
```suggestion
In addition to the nodes allowed at the root of [RAML typed fragments](#typed-fragments), the RAML libraries allow the following optional nodes:
```

File: versions/raml-10/raml-10.md [hunk 8]
```diff
@@ -3206,11 +3212,11 @@ traits:
...
 
 |Name | Description |
 |:--------|:------------|
-| types?<br>schemas?<br>resourceTypes?<br>traits?<br>securitySchemes?<br>annotationTypes?<br>(&lt;annotationName&gt;)?<br>uses? | The definition of each node is the same as that of the corresponding node at the root of a RAML document. A library supports annotation node like any other RAML document.
+| types?<br>schemas?<br>resourceTypes?<br>traits?<br>securitySchemes?<br>annotationTypes?<br>(&lt;annotationName&gt;)? | The definition of each node is the same as that of the corresponding node at the root of a RAML document. A library supports annotation node like any other RAML document.
```
Comment (priyaayyar) [inline versions/raml-10/raml-10.md:3219]:
```suggestion
| types?<br>schemas?<br>resourceTypes?<br>traits?<br>securitySchemes?<br>annotationTypes?<br>(&lt;annotationName&gt;)? | The definition of each node is the same as that of the corresponding node at the root of a RAML document. A library supports the annotation node like any other RAML document.
```

**Review by priyaayyar — CHANGES_REQUESTED** [review]

File: versions/raml-10/raml-10.md (hunk 3 above)
Comment (dukesphere) [inline versions/raml-10/raml-10.md:3077]:
```suggestion
In addition to the nodes allowed in any of the Relevant RAML Specification Sections above, the following OPTIONAL node is allowed:
```

File: versions/raml-10/raml-10.md (hunk 4 above)
Comment (dukesphere) [reply versions/raml-10/raml-10.md:3081]: it's intentional.

File: versions/raml-10/raml-10.md (hunk 5 above)
Comment (dukesphere) [inline versions/raml-10/raml-10.md:3083]:
```suggestion
If a file begins with a RAML fragment identifier line, and the fragment identifier is not a Library, Overlay, or Extension, the contents of the file after the removal of the RAML fragment identifier line and any `uses` node MUST be valid structurally according to the relevant RAML specification. For example, a RAML file beginning with `#%RAML 1.0 Trait` must have the structure of a RAML trait declaration as defined in the [Resource Types and Traits](#resource-types-and-traits) section. Including the file in a correct location for a trait declaration results in a valid RAML file.
```

File: versions/raml-10/raml-10.md (hunk 6 above)
Comment (dukesphere) [reply versions/raml-10/raml-10.md:3085]: I'd leave the original paragraph as-is and decline the suggested change.

File: versions/raml-10/raml-10.md (hunk 7 above)
Comment (dukesphere) [reply versions/raml-10/raml-10.md:3215]: Again, I'd leave as-is and decline the suggested change.

File: versions/raml-10/raml-10.md (hunk 1 above)
Comment (dukesphere) [inline versions/raml-10/raml-10.md:3248]:
```suggestion
You can apply any number of libraries by using the OPTIONAL **uses** node at the root of a RAML or RAML typed fragment file. The value of the **uses** node is a map of key-value pairs. The keys are treated as library names, or namespaces, and the value MUST be the location of a RAML library file, usually an external RAML library fragment document.
```

File: versions/raml-10/raml-10.md (hunk 2 above)
Comment (dukesphere) [reply versions/raml-10/raml-10.md:3252]: I'd leave the original as-is and decline the suggested change.

**Review by dukesphere — APPROVED** [review]

**Review by priyaayyar — APPROVED** [review]

**Pull request merged**